db/migrate/20160113120732_dedupe_upps.rb

```diff
@@ -0,0 +1,19 @@
+class DedupeUpps < ActiveRecord::Migration
+  def change
+    UserProjectPreference.transaction do
+      UserProjectPreference.connection.execute <<-SQL
+        DELETE FROM user_project_preferences
+        WHERE id IN (
+          SELECT id
+          FROM (
+            SELECT id, ROW_NUMBER() OVER (PARTITION BY user_id, project_id ORDER BY id ASC) AS row
+            FROM user_project_preferences
+          ) dupes
+          WHERE dupes.row > 1
+        );
+      SQL
+
+      add_index :user_project_preferences, [:user_id, :project_id], unique: true
```

Comment:
I think we can get rid of the project one, but the user one we might want to keep because of https://github.com/zooniverse/Panoptes/blob/master/app/controllers/emails_controller.rb#L50